feat: add hide details setting for shared dashboards

zlwaterfield opened the pull request:
## Changes

Introduces a 'Hide insights menu' switch in the sharing modal, allowing users to hide creator, last edit, and filter details in shared dashboards. Updates backend to filter out PII fields when this option is enabled and ensures the frontend respects the setting when rendering dashboard items.

Waiting on https://github.com/PostHog/posthog/pull/35761.

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

zlwaterfield commented:
I'll rebase this after https://github.com/PostHog/posthog/pull/35761 is merged.

zlwaterfield commented on posthog/api/insight.py:
```diff
@@ -710,6 +710,13 @@ def to_representation(self, instance: Insight):
         representation["filters_hash"] = self.insight_result(instance).cache_key
 
+        # Hide PII fields when hideExtraDetails from SharingConfiguration is enabled
+        if self.context.get("hide_extra_details", False):
```
should we also not return query so the filter information isn't accessible?

aspicer replied:
I was just coming here to comment on that - there's a ton of information in the insight, especially if they have hogql queries - it can leak their tables, their queries, their filters, etc.

However, I think it's probably rather fraught to go through each insight type and figure out what you can remove that doesn't impact how it is displayed but hides potentially private info and is probably out of scope for trying to get this out.

Probably would avoid for now - if you wanted to design a framework for removing that information (probably needs to be in the query runners themselves because it matters what type of insight it is), you could, but would be okay punting. I'd leave a comment mentioning this is a concern here.

thmsobrmlr replied:
The query is required to render the insight at the moment. It should also be available from the insight itself already. So suggest not removing that here.

aspicer commented on frontend/src/lib/components/Sharing/SharingModal.tsx:
```diff
@@ -238,6 +238,27 @@ export function SharingModalContent({
+                                                        bordered
+                                                        label={
+                                                            <div className="flex items-center">
+                                                                <span>Hide details panel</span>
```
Maybe just "Hide insight details" instead of "panel"?

zlwaterfield committed:
Add 'hideExtraDetails' sharing option for insights

Introduces a new 'hideExtraDetails' setting to sharing configurations, allowing users to hide PII and extra insight details (such as creator and filters) when sharing dashboards or insights. Updates frontend components, schema definitions, and backend serializers to support this option.

zlwaterfield committed:
Merge branch 'master' into zach/shared-dashboard-hide-details

zlwaterfield committed:
Update label text in SharingModal details panel

Changed the label from 'Show details panel' to 'Show insight details' for improved clarity in the SharingModal component.

zlwaterfield committed:
Fix tooltip text for insight details toggle

Corrected the tooltip to state that viewers won't see extra insight details when the option is disabled, clarifying the behavior for users.

zlwaterfield committed:
Fix context key for hiding extra details in sharing API

Renamed the context key from 'hide_details' to 'hide_extra_details' to ensure consistency and correct behavior when hiding PII-related information in the SharingViewerPageViewSet.

github-actions[bot] committed:
Update UI snapshots for `chromium` (16)

github-actions[bot] committed:
Update UI snapshots for `chromium` (16)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield committed:
Update test_sharing_configuration_model.py
